pandas/io/parsers.py

```diff
@@ -1387,6 +1390,20 @@ def _wrap_compressed(f, compression, encoding=None):
             data = bz2.decompress(f.read())
             f = StringIO(data)
         return f
+    elif compression == 'zip':
```

Comment:
This is not that trivial. `wrap_compressed()` handles `file objects`. I'd have to refactor this code to handle for this. See the logic below 
```
        if isinstance(f, compat.string_types):
            f = _get_handle(f, 'r', encoding=self.encoding,
                            compression=self.compression)
        elif self.compression:
            f = _wrap_compressed(f, self.compression, self.encoding)
```
I would then have to modify `_get_handle` to deal with `file objects` as well as `strings`